README.md

```diff
@@ -9,57 +9,314 @@
 
 > ILP virtual ledger plugin for directly transacting connectors
 
-### NOTE:
-
-The current version of `ilp-plugin-virtual` is designed to be used in [ILP
-Kit](https://github.com/interledgerjs/ilp-kit). An HTTP server is required in
-order to pass RPC calls to the plugin, and to perform authorization on RPC calls.
-
 ## Installation
 
 ``` sh
 npm install --save ilp-plugin-virtual
 ```
 
-## Usage
+# ILP Plugin Payment Channel Framework
 
-You can test out the ledger plugin by running `npm test` on your machine.  To
-include debug information during the test, run `DEBUG=ilp* npm test`.
+ILP Plugin virtual exposes a field called `MakePluginVirtual`.  This function
+takes a [Payment Channel Backend](#payment-channel-backend-api), and returns a
+LedgerPlugin class.
 
-When instantiating the plugin, make sure to pass in an object with the right fields: 
+- [Example Code (w/ Claims)](#example-code-with-claim-based-settlement)
+- [Example Code (w/ Payments)](#example-code-with-unconditional-payment-based-settlement)
+- [Extended Plugin Backend API](#backend-api-with-extensions-for-payment-channels)
+- [Payment Channel Backend API](#payment-channel-backend-api)
+- [Plugin Context API](#plugin-context-api)
+
+## Example Code with Claim-Based Settlement
 
 ```js
-{
-  "maxBalance": "highest balance allowed",
-  "prefix": "prefix for ilp address",
-  "secret": "your secret. your peer must know the corresponding public key.",
-  "peerPublicKey": "base64url encoded public key of your peer",
-  "rpcUri": "https://example.com/rpc", // this is used to communicate with your peer (see below)
-  "_store": { /* this object will be created by the connector */ },
-  "info": { /* the object to be returned by getInfo */ }
-}
-```
+const { MakePluginVirtual } = require('ilp-plugin-virtual')
+const { NotAcceptedError } = require('ilp-plugin-shared').Errors
+const Network = require('some-example-network')
+const BigNumber = require('bignumber.js')
 
-# Receiving RPC calls
+return MakePluginVirtual({
+  connect: async function (ctx, opts) {
+    await Network.connectToNetwork()
 
-In order to use plugin virtual, you need an HTTP endpoint that passes calls to the plugin.
-The url must take one query parameter (`method`), as well as a JSON body. The method parameter
-and parsed body must be passed to the plugin like so:
+    ctx.state.bestClaim = ctx.backend.getMaxValueTracker('incoming_claim')
+    ctx.state.maxInFlight = opts.maxInFlight
+  },
 
-```js
-plugin.receive(method, body)
-```
+  handleIncoming: async function (ctx, transfer) {
+    const incoming = await ctx.transferLog.getHighestIncomingBalance()
+    const bestClaim = await ctx.state.bestClaim.getMax() || { value: '0', data: null }
 
-# Generate a trustline prefix
+    const exceeds = new BigNumber(incoming)
+      .sub(bestClaim.value)
```

Comment:
addIfNotOverMaximum already occurs to the balance before handleIncoming is called. All the required validation has passed already and the incoming balance is strictly increasing, so this ensures that a transfer that's too high can never be applied